code/datums/brain_damage/imaginary_friend.dm

```diff
@@ -52,27 +52,27 @@
 	var/mob/living/carbon/owner
 	var/datum/brain_trauma/special/imaginary_friend/trauma
 
-/mob/living/mental/friend/Login()
+/mob/abstract/mental/friend/Login()
 	..()
 	to_chat(src, "<span class='notice'><b>You are the imaginary friend of [owner]!</b></span>")
 	to_chat(src, "<span class='notice'>You are absolutely loyal to your friend, no matter what.</span>")
 	to_chat(src, "<span class='notice'>You cannot directly influence the world around you, but you can see what [owner] cannot.</span>")
 
-/mob/living/mental/friend/Initialize(mapload, _trauma)
+/mob/abstract/mental/friend/Initialize(mapload, _trauma)
 	. = ..()
 	trauma = _trauma
 	owner = trauma.owner
 	var/gender = pick(MALE, FEMALE)
 	real_name = owner.species.get_random_name(gender)
 	name = real_name
 	var/list/candidates = list()
-	for(var/mob/living/L in world)
+	for(var/mob/living/L in living_mob_list)
```

Comment:
Should we limit that to mobs of the same species ?
-> Since we are giving the imag. friend a random name of the owners species.

Reply:
Take in mind it can also take robots and simple animals. which I guess would also make sense for a imaginary friend.